### PR TITLE
Added initial implementation for tojson filter (#142)

### PR DESCRIPTION
**Pull request by vvish**
* Implemented serialization to json based or rapidjson framework.

* Added initial filter implementation + tests.

* Entry added to the list of supported filters in README.

### CODE REVIEW

**Comment by flexferrum** [comment]
```
{"intValue":3,"map":{"str1":1},"boolFalseValue":false,"stringValue":"rain","doubleValue":12.123000144958496,"wstringValue":"rain","boolTrueValue":true,"listValue":[1,2,3]}
1: D:\a\Jinja2Cpp\Jinja2Cpp\test\test_tools.h(148): error: Expected equality of these values:
1:   expectedResult
1:     Which is: "{\"map\":{\"str1\":1},\"listValue\":[1,2,3],\"boolFalseValue\":false,\"boolTrueValue\":true,\"wstringValue\":\"rain\",\"stringValue\":\"rain\",\"doubleValue\":12.123000144958496,\"intValue\":3}"
1:   result
1:     Which is: "{\"intValue\":3,\"map\":{\"str1\":1},\"boolFalseValue\":false,\"stringValue\":\"rain\",\"doubleValue\":12.123000144958496,\"wstringValue\":\"rain\",\"boolTrueValue\":true,\"listValue\":[1,2,3]}"
1: Narrow version
1: {"intValue":3,"map":{"str1":1},"boolFalseValue":false,"stringValue":"rain","doubleValue":12.123000144958496,"wstringValue":"rain","boolTrueValue":true,"listValue":[1,2,3]}
1: D:\a\Jinja2Cpp\Jinja2Cpp\test\test_tools.h(148): error: Expected equality of these values:
1:   expectedResult
1:     Which is: L"{\"map\":{\"str1\":1},\"listValue\":[1,2,3],\"boolFalseValue\":false,\"boolTrueValue\":true,\"wstringValue\":\"rain\",\"stringValue\":\"rain\",\"doubleValue\":12.123000144958496,\"intValue\":3}"
1:   result
1:     Which is: L"{\"intValue\":3,\"map\":{\"str1\":1},\"boolFalseValue\":false,\"stringValue\":\"rain\",\"doubleValue\":12.123000144958496,\"wstringValue\":\"rain\",\"boolTrueValue\":true,\"listValue\":[1,2,3]}"
1: Wide version
```
`rapidjson` reorders object fields during the serialization. So, actual and expected results doesn't satisfy each other.

**Comment by vvish** [comment]
Yes, it looks like the order depends on the target platform. Builds for Linux pass the tests. I will investigate what causes the difference in the behaviour.

File: src/rapid_json_serializer.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+ValueWrapper::ValueWrapper(rapidjson::Value&& value, std::shared_ptr<rapidjson::Document> document)
+    : m_value(std::move(value))
+    , m_document(document)
+{
+}
+
+std::string ValueWrapper::AsString(const uint8_t indent) const
+{
+    using Writer = rapidjson::Writer<rapidjson::StringBuffer, rapidjson::Document::EncodingType, rapidjson::ASCII<>>;
```
Comment (flexferrum) [inline src/rapid_json_serializer.cpp]: I think it's better to use UTF8 here

File: src/rapid_json_serializer.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    ValueWrapper(ValueWrapper&&) = default;
+    ValueWrapper& operator=(ValueWrapper&&) = default;
+
+    std::string AsString(uint8_t indent = 0) const;
+
+private:
+    ValueWrapper(rapidjson::Value&& value, std::shared_ptr<rapidjson::Document> document);
+
+    rapidjson::Value m_value;
+    std::shared_ptr<rapidjson::Document> m_document;
```
Comment (flexferrum) [inline src/rapid_json_serializer.h]: Why not unique_ptr?

File: src/rapid_json_serializer.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+public:
+    DocumentWrapper();
+
+    DocumentWrapper(DocumentWrapper&&) = default;
+    DocumentWrapper& operator=(DocumentWrapper&&) = default;
+
+    ValueWrapper CreateValue(const InternalValue& value) const;
+
+private:
+    std::shared_ptr<rapidjson::Document> m_document;
```
Comment (flexferrum) [inline src/rapid_json_serializer.h]: Why not unique_ptr as well?

File: src/rapid_json_serializer.h (hunk 2 above)
Comment (vvish) [reply src/rapid_json_serializer.h]: If I understand that correctly from the code, the rapidjson::Document owns the allocator that is used by values. So the lifetime of the Values created with this allocator shouldn't exceed the lifetime of the Document.

**Pull request merged**